File: USACO/Training/1.1/ride.py
"""
ID: xuethan1
LANG: PYTHON3
TASK: ride
"""

import string
import math
import itertools


def convert_arr_int(arr):
    new_arr = []
    for e in arr:
        new_arr.append(int(e))
    return new_arr


def find_letter(l) -> int:
    alpha = list(string.ascii_uppercase)
    for i in range(26):
        if alpha[i] == l:
            return i+1
    return -1


def main():
    with open("ride.in", "r") as input_file:
        input_data = input_file.read()
    line1 = input_data.split("\n")[0]
    line2 = input_data.split("\n")[1]

    line1_total = 1
    line2_total = 1

    for i in range(len(line1)):
        print(find_letter(line1[i]))
        line1_total *= find_letter(line1[i])
    for i in range(len(line2)):
        print(find_letter(line2[i]))
        line2_total *= find_letter(line2[i])

    print(line1_total)
    print(line2_total)

    output_file = open("ride.out", "w")
    if line1_total % 27 == line2_total % 27:
        output_file.write("GO")
    else:
        output_file.write("STAY")
    output_file.close()


if __name__ == "__main__":
    main()

File: USACO/Training/1.1/ride.java
// This template code suggested by KT BYTE Computer Science Academy
//   for use in reading and writing files for USACO problems.

// https://content.ktbyte.com/problem.java

/*
ID: xuethan1
LANG: JAVA
TASK: ride
*/

import java.util.*;
import java.io.*;

public class ride {
  public static void main(String[] args) throws FileNotFoundException {
    char[] alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".toCharArray();



    Scanner in = new Scanner(new File("ride.in"));
    ArrayList<String> lines = new ArrayList<String>();
    while(in.hasNextLine()){
      lines.add(in.nextLine());
    }
    in.close();

    char[] line1 = lines.get(0).toCharArray();
    char[] line2 = lines.get(1).toCharArray();

    int line1_total = 1;
    int line2_total = 1;

    // System.out.println("=========");

    for(int i = 0; i<line1.length; i++){
      // System.out.println(find_letter(line1[i], alphabet));
      line1_total *= find_letter(line1[i], alphabet);
    }
    // System.out.println("=========");

    for(int i = 0; i<line2.length; i++){
      // System.out.println(find_letter(line2[i], alphabet));
      line2_total *= find_letter(line2[i], alphabet);
    }

    // System.out.println("totals: ");
    // System.out.println(line1_total);
    // System.out.println(line2_total);

    PrintWriter out = new PrintWriter(new File("ride.out"));

    if(line1_total % 47 == line2_total % 47){
      out.println("GO");
    }
    else{
      out.println("STAY");
    }

    out.close();
  }

  static int find_letter(char l, char[] alpha){
    for (int i = 0; i<alpha.length; i++){
      if(alpha[i] == l){
        return i+1;
      }
    }
    return -1;
  }
}
